File: examples/01-kerr-spacetime/kerr-orbits.py
from __future__ import division

import sys
from math import *
import numpy as np
import matplotlib.pyplot as plt

# import SIM5 library
sys.path.append('../../..')
import sim5

# make array of spin values from 0 to 1
bh_spin = np.linspace(0, 1, 200)

# compute 
# - black-hole event horizon radius,
# - marginally stable orbit radius,
# - marginally bound radius,
# - photon orbit radius
# for all spins in bh_spin array
rbh = map(sim5.r_bh, bh_spin)
rms = map(sim5.r_ms, bh_spin)
rmb = map(sim5.r_mb, bh_spin)
rph = map(sim5.r_ph, bh_spin)

# make a plot of rbh and rms as a function of spin
plt.title("Kerr orbits")
plt.plot(bh_spin, rms, label='marg. stable orbit')
plt.plot(bh_spin, rmb, label='marg. bound orbit')
plt.plot(bh_spin, rph, label='photon orbit')
plt.plot(bh_spin, rbh, label='event horizon')
plt.xlabel(r'BH spin')
plt.ylabel(r'radius [$r_{\rm g}$]')
plt.legend()

plt.show()

File: examples/01-kerr-spacetime/kerr-orbits.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include "sim5lib.h"


int main(int argc, char *argv[])
//***************************************************
{
    double a;
    double rbh, rph, rmb, rms;

    // print header
    fprintf(stdout, "# Locations of black-hole horizon (r_bh), photon orbit radius (r_ph),\n");
    fprintf(stdout, "# marginally bound orbit (r_mb) and marginally stable orbit (r_ms)\n");
    fprintf(stdout, "# in Kerr spacetime as a function of black-hole spin.\n");
    fprintf(stdout, "# Line format: spin  r_bh  r_ph  r_mb  r_ms\n");
    fprintf(stdout, "# ----\n");

    // go over spins from 0 to 1
    for (a=0.0; a<1.0; a+=0.01) {
        // get values of different radii
        rbh = r_bh(a);
        rph = r_ph(a);
        rmb = r_mb(a);
        rms = r_ms(a);

        // print it out
        fprintf(stdout, "%.4f  %e  %e  %e  %e\n", a, rbh, rph, rmb, rms);
    }

    return 0;
}

